metabase/metabase
# Viz Card Settings in popups

Epic: https://github.com/metabase/metabase/issues/24840

Changes Column Formatting settings to be displayed in a popup rather than take over the Cart Settings side panel. While the actual rendering of the settings hasn't changed, the location where they are rendered is different. I made a new component to handle rendering the settings in a popup, and updated the following components to pass a ref to anchor the popup to:
- `ChartSettingFieldPicker`
- `ChartSettingFieldPartition`
- `ChartSettingOrderedColumns`
- `ChartSettingsListColumns`

They're the only widgets that call `onShowWidget`, so they should be the only ones that need to be updated.

## How to test

- New Question - > Sample Dataset -> People -> Visualize
- Go to settings (bottom right of screen)
- Under columns, click the ellipsis for Birth Date
- Change the date style

![image](https://user-images.githubusercontent.com/1328979/187775796-41dd9e50-4a3c-45d3-87e2-e4b78c6f6ddc.png)

This is one example, other settings should render in the same way for other types of visualizations as well.
